samza-core/src/main/java/org/apache/samza/config/SerializerConfig.java

```diff
@@ -0,0 +1,108 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.config;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.apache.samza.SamzaException;
+import org.apache.samza.serializers.ByteBufferSerdeFactory;
+import org.apache.samza.serializers.ByteSerdeFactory;
+import org.apache.samza.serializers.DoubleSerdeFactory;
+import org.apache.samza.serializers.IntegerSerdeFactory;
+import org.apache.samza.serializers.JsonSerdeFactory;
+import org.apache.samza.serializers.LongSerdeFactory;
+import org.apache.samza.serializers.SerializableSerdeFactory;
+import org.apache.samza.serializers.StringSerdeFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Provides helpers for accessing configs related to serialization and deserialization.
+ */
+public class SerializerConfig extends MapConfig {
+  public static final Logger LOGGER = LoggerFactory.getLogger(SerializerConfig.class);
+
+  public static final String SERIALIZER_PREFIX = "serializers.registry.%s";
+  public static final String SERDE_FACTORY_CLASS = SERIALIZER_PREFIX + ".class";
+  public static final String SERIALIZED_INSTANCE_SUFFIX = ".samza.serialized.instance";
+  public static final String SERDE_SERIALIZED_INSTANCE = SERIALIZER_PREFIX + SERIALIZED_INSTANCE_SUFFIX;
+
+  public SerializerConfig(Config config) {
+    super(config);
+  }
+
+  /**
+   * Returns the pre-defined serde factory class name for the provided serde name. If no pre-defined factory exists,
+   * throws an exception.
+   */
+  public static String getSerdeFactoryName(String serdeName) {
+    String serdeFactoryName = doGetSerdeFactoryName(serdeName);
+    LOGGER.info(String.format("Using default serde %s for serde name %s", serdeFactoryName, serdeName));
+    return serdeFactoryName;
+  }
+
+  /**
+   * @param name name of the serde in the config
+   * @return serde factory class name for {@code name}; empty if no class was found in the config
+   */
+  public Optional<String> getSerdeClass(String name) {
+    return Optional.ofNullable(get(String.format(SERDE_FACTORY_CLASS, name), null));
+  }
+
+  /**
+   * Useful for getting individual serializers.
+   * @return a list of all serializer names from the config file
+   */
+  public List<String> getSerdeNames() {
+    List<String> results = new ArrayList<String>();
+    Config subConfig = subset(String.format(SERIALIZER_PREFIX, ""), true);
+    for (String key : subConfig.keySet()) {
+      if (key.endsWith(".class")) {
+        results.add(key.replace(".class", ""));
+      }
+    }
+    return results;
+  }
+
+  private static String doGetSerdeFactoryName(String serdeName) {
+    switch (serdeName) {
+      case "byte":
+        return ByteSerdeFactory.class.getCanonicalName();
```

Comment:
Use 'Class.getName()' for class names to be used for classloading. Canonical names may not be unique within the classloader. E.g., https://stackoverflow.com/questions/15202997/what-is-the-difference-between-canonical-name-simple-name-and-class-name-in-jav

Reply:
Sounds good. The previous Scala code used `getCanonicalName`, but seems reasonable to just make the change here.